### PR TITLE
Proc for generating map renders

### PR DESCRIPTION
**Pull request by clusterfack**
I mean what else is there to say.

It doesn't make it into one large icon unfortunately but someone could do that. Or you could use the goon method to display them to players, shown here http://goonhub.com/maps/cogmap.

Samples
http://puu.sh/pabIB/1325196400.png
http://puu.sh/pabJ1/2e3e94dfd6.png
http://puu.sh/pabJF/761883be9a.png

Note space is left as white, if map renders are displayed to users through the goon's method, this is fine as you can underlay the spacebase I have included. 

If you want the precise space turfs you can make a minor modification to the proc to cause them to render, though it severely bloats the runtime of map rendering.

### CODE REVIEW

**Comment by Probe1** [comment]
:clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap: :clap:

**Comment by Shadowmech88** [comment]
Thank you codeman, you are our greatest ally.

File: maprendering/maprendering.dm [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/client/proc/maprender()
+	set category = "Mapping"
+	set name = "Generate Map Render"
+
+	if(!holder)
```
Comment (PJB3005) [inline maprendering/maprendering.dm:5]: You need to check rights properly.

**Comment by Probe1** [comment]
Please write a fool proof guide on using this for users to come so we don't run into the same deal where no one knows what the fuck to do or how to fix it.

File: maprendering/maprendering.dm [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		to_chat(src, "Only administrators may use this command.")
+		return
+	if(alert("Sure you want to do this? It should NEVER be done in an active round and cannot be cancelled", "generate maps", "Yes", "No") == "No")
```
Comment (optimumtact) [inline maprendering/maprendering.dm:8]: > cannot be cancelled
> 
> one alert box below allowing cancel

truly a paranoid man

Comment (clusterfack) [inline maprendering/maprendering.dm]: For when someone inevitably goes *I wonder what this does* and presses it

File: maprendering/maprendering.dm [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	else if(allz == "Specific")
+		zlevel = input("Input zlevel you wish to render") as num
+
+	message_admins("[ckey]/[src] started rendering maps")
+	log_admin("[ckey]/[src] started rendering maps")
+
+	maprenders(zlevel, allz == "All" ? 1 : 0)
+
+/client/proc/maprenders(var/currentz = 1, var/allz = 0)
+
```
Comment (PJB3005) [inline maprendering/maprendering.dm:25]: Whitespace!

Comment (clusterfack) [inline maprendering/maprendering.dm]: You don't say

**Comment by Probe1** [comment]
image


Double doors go batty.

Engines set their dir to default 1

**Comment by clusterfack** [comment]
Engines is easy, the door thing might be a huge pain and more than its worth

**Comment by Probe1** [comment]
its not that big of a deal to be honest.

File: maprendering/maprendering.dm (hunk 2 above)
Comment (ghost) [inline maprendering/maprendering.dm:8]: I may have done this before.

Comment (PJB3005) [inline maprendering/maprendering.dm:8]: Yep pretty sure he did do nano map gen during a round before.

Let's not forget him to trying to move every turf with SDQL. 

Celt shouldn't have +DEBUG

Comment (ghost) [inline maprendering/maprendering.dm:8]: I shouldn't have any flag to be quite honest

**Comment by clusterfack** [comment]
Yep all done, to update the render just run the generate map render procs, zip up the file it outputs in renderoutput, put it in a dropbox, then let pomf download/extract it and they will be updated.

He has all the files needed for hosting it on his website you can check them out by going to ss13.moe/minimaps/

**Comment by Skullyton** [comment]
So, would this mean if you were to make a modification to a map, you'd run the proc over your current version and leave a link to the zip of all the images in the pull request description or something?

**Comment by clusterfack** [comment]
Well right now only pomf has access to what he hosts, so the only way would be to send it to him. But in the future we can probably work out some system like that. 

Say a dropbox that collaborators have access to that is synced to the hosted files so we can modify them.

**Comment by clusterfack** [comment]
It also only takes ~2 minutes to run for a single map on my computer, which means it doesn't even take that long to update.

**Comment by clusterfack** [comment]
Okay pomf has designed the folder for a dropbox that I have access to, including a zip to an updated map render in your map PR will allow me to update them upon merging.

**Comment by dylanstrategie** [comment]
It would be a bit more handy if the space rendered as transparent instead of whitespace for wiki editing

**Comment by clusterfack** [comment]
Fun fact, it does now!

**Comment by clusterfack** [comment]
Went ahead and fixed multitile objects

File: maprendering/maprendering.dm (hunk 1 above)
Comment (MrsTonedOne) [inline maprendering/maprendering.dm:5]: honestly the check shouldn't exist at all, the only way somebody without the rights could call that is advance proc call, and at that point they already could do so much more damage, including changing their own flags.

**Comment by dylanstrategie** [comment]
Alright, time to give this babe a go and see how the wiki likes it

**Comment by PJB3005** [comment]
Hey

**Comment by gbasood** [comment]
\>It doesn't make it into one large icon unfortunately but someone could do that.
Goon's method makes map slices for the record
Also if someone makes me a web map viewer I will work on webstats (at least until school starts)

**Pull request merged**